src/middleware.js

```diff
@@ -34,56 +34,34 @@ function validateAuthorizationOptions(options = {}) {
 }
 
 // Check to see if an already Authenticated user is Authorized to do something,
-// based on: a) their role; b) arbitrary aspects of the user payload. For
+// based on the `req` and the `user` payload from the access token. For
 // example, if a user must have the 'admin' role, or a user's `sub` claim
 // must match an expected id. NOTE: isAuthorized() assumes (and depends upon)
 // isAuthenticated() being called first.
-function isAuthorized(options) {
-  if (!validateAuthorizationOptions(options)) {
-    throw new Error('invalid authorization options');
+function isAuthorized(authorizeUser) {
+  if (typeof authorizeUser !== 'function') {
+    throw new Error('invalid authorization function');
   }
 
-  const { roles, authorizeUser } = options;
-
   return function (req, res, next) {
-    if (!req.user) {
+    const { user } = req;
+    if (!user) {
       next(createError(401, `no user or role info`));
       return;
     }
 
     // If these checks fail for any reason, return a 403
     try {
-      const { user } = req;
-
-      // If defined, check that all of the expected roles are present in this user's roles
-      if (roles) {
-        if (!user.roles) {
-          next(createError(403, `user missing roles`));
-          return;
-        }
-        for (const role of roles) {
-          if (!user.roles.includes(role)) {
-            next(createError(403, `user missing required role: ${role}`));
-            return;
-          }
-        }
-      }
-
-      // If defined, check that the user's payload data matches what's expected
-      if (authorizeUser) {
-        if (!authorizeUser(user)) {
-          next(createError(403, `user not authorized`));
-          return;
-        }
+      if (!authorizeUser(req, user)) {
```

Comment:
Should we also log the error into the logger for debugging purposes? Just in case we have a scenario where we branch off into the error, when we weren't supposed to.

Reply:
Because we use `next()` all errors will get logged as the 403 is returned.  NOTE: I log in the `catch` case because it's exceptional that it should happen.  This one isn't exceptional, so I don't think we need more than the http 403 logged (i.e., there isn't any more info to add).

Reply:
Ah, I see, cool!